### ISSUE TITLE
Problem including BatchNorm after Dense layers

### ISSUE THREAD

**Issue by bernardohenz** [mozilla/DeepSpeech#1445]
I am trying to include BatchNormalization layers after dense layers, much like the paper DeepSpeech2 implements (next I am going to implement BN on the LSTMs). In summay, I am using the following implementation of batchnorm inside the ```BiRNN``` definition:

```python
    layer_1 = tf.layers.batch_normalization(inputs=layer_1, momentum=FLAGS.batchnorm_momentum, training=training_ph)
```

where ```training_ph``` is a boolean placeholder set to ```True``` during training, and ```False``` otherwise; and ```batchnorm_momentum=0.9```. I've also passing the updateops for session during training (on line https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L1637):

```python
update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
 _, _, current_step, batch_loss, batch_report = session.run([train_op, update_ops ,global_step, loss, report_params], **extra_params)
```

Now to the problem: it seems that it works on training (loss going down), but it seems bugged in validation. As batchnorm is behaving differently for training and validation, the ```training_ph``` appears to be correct, and I suspect that the params of batchnorm (```moving_mean``` and ```moving_variance```) are not being properly updated.

Ps: I tried to check the branch of 'bnlstm', but it didn't help me to fix this. @reuben where you able to use batchnorm sucessfully?

**Comment by kdavis-mozilla** [comment]
Hey @bernardohenz, while this _is_ interesting and BN is usually hard to tune, as this isn't bug with the current code or a feature request, could you take this discussion to [discourse](https://discourse.mozilla.org/c/deep-speech) where such non-bug non-feature request items are discussed. Thanks.

PS: Last I remember @reuben was also having some problems with batch norm, and unrelatedly I talked to an author of the DeepSpeech2 paper and he mentioned tuning BN is an art.
